src/BCFKeyValueData.cc

```diff
@@ -217,6 +246,51 @@ Status BCFKeyValueData::dataset_bcf(const string& dataset,
     return Status::OK();
 }
 
+// Search all the buckets that may hold records within the query range. The tricky
+// corner case is the bucket immediately before the beginning of the query range.
+// It may hold records that start outside the range, but end inside it. We want
+// to return those as well.
+//
+// Return value:
+// - If none of buckets exist, return NotFound.
+// - If at least one of the buckets exist, but no records were found, return an empty list.
+// - Otherwise, return the list of records that overlap with the query range.
```

Comment:
We should return Status::NotFound only in some erroneous condition. Zero results for a query is not an error. Exhibiting different behavior based on whether certain internal buckets exist or not is an abstraction violation.